demos/mqtt/subscription-manager/mqtt_subscription_manager.c

```diff
@@ -379,7 +379,7 @@ SubscriptionManagerStatus_t SubscriptionManager_RegisterCallback( const char * p
     assert( topicFilterLength != 0 );
     assert( callback != NULL );
 
-    SubscriptionManagerStatus_t returnStatus = SUBSCRIPTION_MANAGER_INVALID;
+    SubscriptionManagerStatus_t returnStatus;
```

Comment:
Ideally, the status value should be initialized with a failure value, however, no value in the `SubscriptionManagerStatus_t` represents a general failure.
As the `SUBSCRIPTION_MANAGER_SUCCESS` has a value of **1**, the risk of default initialization causing an false success return value, is mitigated.